### PR TITLE
NO-TICKET: use actual exit code returned by validator reactor

### PR DESCRIPTION
**Pull request by Fraser999**
This PR fixes an issue where the exit code returned by the validator reactor is ignored.

It also borrows code from #1070 which ensures all destructors are run before exiting `main`.

### CODE REVIEW

File: node/src/app/cli.rs [hunk 1]
```diff
@@ -173,8 +172,7 @@ impl Cli {
                 // initializer2_runner.run(&mut rng).await;
 
                 match initializer_runner.run(&mut rng).await {
-                    ReactorExit::ProcessShouldExit(ExitCode::Success) => return Ok(()),
-                    ReactorExit::ProcessShouldExit(exit_code) => process::exit(exit_code as i32),
```
Comment (goral09) [inline node/src/app/cli.rs:177]: What was `process::exit` doing then? If we don't need it anymore.

File: node/src/app/cli.rs [hunk 2]
```diff
@@ -173,8 +172,7 @@ impl Cli {
...
+                    ReactorExit::ProcessShouldExit(exit_code) => return Ok(exit_code as i32),
```
Comment (goral09) [inline node/src/app/cli.rs:175]: Are we going to log the exit code anywhere? If not, then maybe add it to the `cli.rs`.

**Review by goral09 — APPROVED** [review]
Please log the `ExitCode` somewhere in the node.

File: node/src/app/cli.rs (hunk 1 above)
Comment (Fraser999) [reply node/src/app/cli.rs:177]: It exits the process immediately without running destructors.  This PR uses Marc's approach of passing the exit code out to `main` so we can drop everything and only then call `process::exit`.

**Comment by goral09** [comment]
Once this is merged, let's port it to 0.9.2 and 0.9.3 release branches.

File: node/src/app/cli.rs (hunk 2 above)
Comment (Fraser999) [reply node/src/app/cli.rs:175]: Done in [ad76d63](https://github.com/CasperLabs/casper-node/pull/1125/commits/ad76d638d8701e54d764214ec0bcba7463f8134f).

**Comment by Fraser999** [comment]
bors merge

**Review by afck — APPROVED** [review]

**Comment by Fraser999** [comment]
bors merge-

**Comment by Fraser999** [comment]
bors merge

**Comment by Fraser999** [comment]
bors merge-

**Comment by Fraser999** [comment]
bors merge

**Comment by goral09** [comment]
Can you create PR against `release-0.9.3`?

**Comment by Fraser999** [comment]
bors merge

**Pull request merged**